packages/neon/neon/lib/src/models/account.dart

```diff
@@ -2,20 +2,34 @@ import 'dart:convert';
 
 import 'package:collection/collection.dart';
 import 'package:crypto/crypto.dart';
-import 'package:flutter/foundation.dart';
 import 'package:json_annotation/json_annotation.dart';
+import 'package:meta/meta.dart';
 import 'package:nextcloud/nextcloud.dart';
 
 part 'account.g.dart';
 
+/// Credentials interface
+@internal
+@immutable
+abstract interface class Credentials {
+  /// Url of the server
+  abstract final String serverURL;
+
+  /// Username
+  abstract final String username;
+
+  /// Password
+  abstract final String? password;
```

Comment:
Does this need to be nullable?

I don't see any reason for it (no code seems to break with it being changed but I might be wrong)

Reply:
At some point we will have guest accounts without passwords :)